x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/writer/FieldConfigWriter.java

```diff
@@ -59,6 +59,7 @@ public FieldConfigWriter(AnalysisConfig config, Set<MlFilter> filters, List<Sche
     /**
      * Write the Ml autodetect field options to the outputIndex stream.
      */
+    @SuppressWarnings("unused") // CATEGORIZATION_TOKENIZATION_IN_JAVA seems to be used for performance testing
```

Comment:
L70-73 shows up as dead, but from the comment on the constant it appears this is used occassionaly, so maybe supressing and leaving a comment is a good idea.

Reply:
Instead of having this constant, could the if condition be commented out with a comment it should be uncommented when needed for debugging? If code must be changed to enable this, then it seems a constant isn't really helping anything here, and adding the unused suppression adds error room for other bugs.

Reply:
@droberts195 can probably comment. As far as I see the CATEGORIZATION_TOKENIZATION_IN_JAVA is used in several places throughout the ML test code, but needs to be enabled manually. I thing supressing the warning and commenting here is sufficient

Reply:
Yes, it's used in 9 places.  The alternative would be to get rid of the constant and comment out the code in the 9 places, but then somebody who wanted to run the test would have to remember all 9 places to uncomment.  It's nice to just be able to change 1 constant.

I guess the scope of suppression could be minimised by adding a new method `writeCategorizationFilters` and just applying the suppression of warnings to that new method.  `writeScheduledEvents` is already broken into a separate method that checks a (different) condition and maybe writes one piece of config.